tests/test_pre_processing.py

```diff
@@ -1,14 +1,317 @@
+        self.package_path = self._package_path
+
+    def test_annuity_empty_no_cost_params_raises_error(self):
+        """
+        Tests that a value error is raised when the annuity is left empty and the
+        other cost parameters are not included.
+        """
+        # copy scenario csv file to the package path
+        f_name = "annuity_empty_no_cost_params.csv"
+        shutil.copy(os.path.join(self.test_inputs_pre_p, f_name), os.path.join(self.package_path, f_name))
+        # check if calling the pre_processing function with the given scenario raises a value error
+        with pytest.raises(ValueError):
+            pre_processing(self.pre_p_dir, wacc=1)
+
+    def test_annuity_defined_no_cost_params_uses_annuity(self):
+        """
+        Tests that the annuity is directly used when it is defined and the other cost
+        parameters are not included.
+        """
+        # copy scenario csv file to the package path
+        f_name = "annuity_defined_no_cost_params.csv"
+        shutil.copy(os.path.join(self.test_inputs_pre_p, f_name), os.path.join(self.package_path, f_name))
+        # call pre_processing function
+        pre_processing(self.pre_p_dir, wacc=1)
+        # check if the actual value matches the expected value
+        expected_value = 107265
+        df = pd.read_csv(os.path.join(self.package_path, f_name), delimiter=';')
+        actual_value = df['capacity_cost'].iloc[0]
+        assert actual_value == expected_value
+
+    def test_annuity_defined_no_cost_params_logs_message(self, caplog):
+        """
+        Tests that an info message is logged when the annuity is directly used and the other cost
+        parameters are not included.
+        """
+        # copy scenario csv file to the package path
+        f_name = "annuity_defined_no_cost_params.csv"
+        shutil.copy(os.path.join(self.test_inputs_pre_p, f_name), os.path.join(self.package_path, f_name))
+        # check if the info message is logged when the pre_processing function is called
+        pre_processing(self.pre_p_dir, wacc=1)
+        assert any(record.levelname == "INFO" and "The annuity cost is directly used for" in record.message
+                   for record in caplog.records)
+
+    def test_annuity_empty_partial_cost_params_raises_error(self):
+        """
+        Tests that a value error is raised when the annuity is left empty and not all other cost
+        parameters are included.
+        """
+        # copy scenario csv file to the package path
+        f_name = "annuity_empty_partial_cost_params.csv"
+        shutil.copy(os.path.join(self.test_inputs_pre_p, f_name), os.path.join(self.package_path, f_name))
+        # check if calling the pre_processing function with the given scenario raises a value error
+        with pytest.raises(ValueError):
+            pre_processing(self.pre_p_dir, wacc=1)
+
+    def test_annuity_defined_partial_cost_params_raises_error(self):
```

Comment:
should it raise an error in that case? why not simply use the provided annuity and issue a warning?

Reply:
I originally made it raise an error because if some but not all of the cost parameters are defined, the post-processing results can't be calculated properly, but I agree with you to raise a warning here and then deal with this problem in post-processing.